cmd/erasure-object.go

```diff
@@ -893,10 +901,14 @@ func (er erasureObjects) DeleteObject(ctx context.Context, bucket, object string
 	goi, gerr := er.GetObjectInfo(ctx, bucket, object, opts)
 	if gerr != nil && goi.Name == "" {
 		switch gerr.(type) {
+		case ObjectNotFound, VersionNotFound:
+			// Continue deleting since the object/version not found
+			// could also mean a dangling object in the storage
 		case InsufficientReadQuorum:
```

Comment:
the reason we don't proceed here because @vadmeste is an optimization to avoid hold locks on objects which have been deleted or are not readable. 

I believe you should bring code such that where we can safely delete the content right after `isObjectDangling()` such that we don't need to proceed to any further delete operations instead delete it right in GetObjectInfo()